Fix create command flags

vdemeester opened the pull request:
> Any command that expects extra flags after positional args needs to set `flags.SetInterspersed(false)`.

Fix #23329 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

thaJeztah commented:
Looks like that test you modified is failing:

```
11:35:24 ----------------------------------------------------------------------
11:35:24 FAIL: docker_cli_create_test.go:22: DockerSuite.TestCreateArgs
11:35:24 
11:35:24 docker_cli_create_test.go:58:
11:35:24     c.Fatalf("Unexpected args. Expected %v, received: %v", expected, cont.Args)
11:35:24 ... Error: Unexpected args. Expected [arg1 arg2 arg with space], received: [arg1 arg2 arg with space -c flags]
11:35:24 
11:35:26 
```

thaJeztah commented:
LGTM otherwise

vdemeester committed:
Fix create command flags

Any command that expects extra flags after positional args needs to set flags.SetInterspersed(false).

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
Oups lol :joy: Updated the expected :angel:

dnephin commented:
Thanks for fixing!

LGTM

jhowardmsft commented:
LGTM

vdemeester commented:
userns and win2lin green but did not notify 😅
